### PR TITLE
fix: textarea invalid icon

### LINKED ISSUE #16355: [Bug]: Error icon appears outside of the textarea element when user passes `cols` property
### Package

@carbon/react, @carbon/styles

### Browser

Chrome, Firefox, Edge

### Package version

1.56.0

### Description

`Error icon` appears outside of the textarea element when user passes `cols` property to TextArea component.

### Screenshots

image

### Steps to reproduce

1. Pass `cols` prop in Textarea component
2. Set `invalid` prop to `true`

Issue can be observed in storybook here: https://react.carbondesignsystem.com/?path=/story/components-textarea--playground&args=invalid:!true;cols:60

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### LINKED ISSUE #16649: [Bug]: Warning icon appears in the middle of the textarea element instead of right upper corner
### Package

@carbon/styles

### Browser

Chrome

### Package version

v1.55.0

### Description

The warning icon is vertically centered within the textarea element instead of be placed in the right upper corner as the invalid icon. Please notice that this is happening in Carbon Angular, not Carbon React.

### Screenshots

Screenshot 2024-06-04 at 10 07 52
Screenshot 2024-06-04 at 10 08 44

### Steps to reproduce

Get to Storybook, [TextArea component ](https://angular.carbondesignsystem.com/?path=/story/components-input-text-area--basic&args=warn:!true), and set the warn directive to true.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by riddhybansal**
Closes #16355 
Closes https://github.com/carbon-design-system/carbon/issues/16649

Error icon appears outside of the textarea element when user passes cols property to TextArea component

#### Changelog

**Changed**

- Some styles.

#### Testing / Reviewing

Pass cols prop in Textarea component
Set invalid prop to true, The icon should lie in the textarea itself

### CODE REVIEW

File: packages/react/src/components/TextArea/TextArea.tsx [hunk 1]
```diff
@@ -323,11 +323,17 @@ const TextArea = React.forwardRef((props: TextAreaProps, forwardRef) => {
...
+    `${prefix}--text-area__wrapper`,
+    {
+      [`${prefix}--text-area__wrapper--cols`]: other.cols,
+    },
+    {
+      [`${prefix}--text-area__wrapper--readonly`]: other.readOnly,
+      [`${prefix}--text-area__wrapper--warn`]: warn,
+      [`${prefix}--text-area__wrapper--slug`]: slug,
+    }
+  );
```
Comment (alisonjoseph) [inline packages/react/src/components/TextArea/TextArea.tsx]:
```suggestion
  const textAreaWrapperClasses = classNames(`${prefix}--text-area__wrapper`, {
    [`${prefix}--text-area__wrapper--cols`]: other.cols,
    [`${prefix}--text-area__wrapper--readonly`]: other.readOnly,
    [`${prefix}--text-area__wrapper--warn`]: warn,
    [`${prefix}--text-area__wrapper--slug`]: slug,
  });
```

minor formatting fix

**Review by alisonjoseph — APPROVED** [review]
This LGTM, just a minor formatting update suggestion.

**Comment by riddhybansal** [comment]
> image Hey, a minor thing I noticed, Is it ok that if it shows the number of characters out of the box?

Am not sure of it may be @alina-jacob would be a right person to answer this.

**Comment by alina-jacob** [comment]
Hi @preetibansalui and @riddhybansal, good catch on the word-counter!
Its supposed to re-size based the width of the text area

The[ screenshot one ](https://github.com/carbon-design-system/carbon/pull/16725#issuecomment-2162941847)is a bug, the counter would sit on the right-edge of the text-area (parallel to the lable, above the input field), not in the far right corner

Here the spec for it: [Website link](https://carbondesignsystem.com/components/text-input/style/#default-text-area-structure)
image

**Review by 2nikhiltom — APPROVED** [review]

**Comment by riddhybansal** [comment]
> Hi @preetibansalui and @riddhybansal, good catch on the word-counter! Its supposed to re-size based the width of the text area
> 
> The [screenshot one](https://github.com/carbon-design-system/carbon/pull/16725#issuecomment-2162941847) is a bug, the counter would sit on the right-edge of the text-area (parallel to the lable, above the input field), not in the far right corner
> 
> Here's the spec for it: [Website link](https://carbondesignsystem.com/components/text-input/style/#default-text-area-structure) image

For this , will be creating another issue.

**Pull request merged**